controllers/front/Payment.php

```diff
@@ -412,4 +421,94 @@ public function buildRecurringData($request = array(), $payload)
         return $request;
     }
 
+    private function buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request = array())
+    {
+        $request['paymentMethod']['type'] = $paymentType;
+        if (!empty($paymentIssuer)) {
+            $request['paymentMethod']['issuer'] = $paymentIssuer;
+        }
+
+        return $request;
+    }
+
+    private function buildReturnURL($request)
+    {
+        $request['returnUrl'] = $this->context->link->getModuleLink(
+            $this->module->name,
+            'Result',
+            array(),
+            $this->ssl
+        );
+
+        return $request;
+    }
+
+    private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIssuer)
+    {
+        $request = array();
+        $request = $this->buildBrowserData($request);
+        $request = $this->buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request);
+        $request = $this->buildPaymentData($request);
+        $request = $this->buildMerchantAccountData($request);
+        $request = $this->buildRecurringData($request, $_REQUEST);
+        $request = $this->buildReturnURL($request);
+
+        // call adyen library
+        /** @var Adyen\PrestaShop\service\Checkout $service */
+        $service = ServiceLocator::get('Adyen\PrestaShop\service\Checkout');
+
+        try {
+            $response = $service->payments($request);
+        } catch (\Adyen\AdyenException $e) {
+            $this->helperData->adyenLogger()->logError(
+                "There was an error with the payment method. id:  " . $cart->id . " Response: " . $e->getMessage()
+            );
+
+            $this->ajaxRender(
+                $this->helperData->buildControllerResponseJson(
+                    'error',
+                    [
+                        'message' => "There was an error with the payment method, please choose another one."
+                    ]
+                )
+            );
+            return;
+        }
+
+        switch ($response['resultCode']) {
+            case 'RedirectShopper':
```

Comment:
Do we need a Default case to non-handle(error) the other flows? (I can only think of PresentToShopper at the moment)

Reply:
I will add that